# Study 832 — High-Yield Credit Momentum — the teardown

The discrimination Newey-West *t*, the 1,000-draw label-shift placebo, the two-era cut, the costed SPY↔IEF timer vs buy-and-hold, and the 20-seed synthetic positive control.

In [1]:
R = {'start': '2007-05-01', 'end': '2026-06-30', 'n_rows': 4822, 'fingerprint': '3c027b47d765', 'L': 126, 'n_days': 4695, 'on_frac': 0.636, 'on_bps': 2.99, 'off_bps': 4.6, 'diff_bps': -1.62, 't_nw': -0.37, 'welch_t': -0.32, 'diff3_bps': -2.55, 't3_nw': -0.61, 'placebo_obs': -1.62, 'placebo_mean': 0.26, 'placebo_sd': 4.67, 'placebo_p': 0.646, 'placebo_sigma': -0.4, 'placebo_draws': 1000, 'era_early_diff': 4.8, 'era_early_t': 0.81, 'era_early_n': 2310, 'era_late_diff': -12.27, 'era_late_t': -1.81, 'era_late_n': 2385, 'n_switches': 201, 'gross_sharpe': 0.645, 't1_net_sharpe': 0.627, 't1_bh_sharpe': 0.619, 't1_net_cagr': 7.32, 't1_bh_cagr': 10.87, 't1_net_dd': -41.3, 't1_bh_dd': -54.7, 't1_active_bps': -1.76, 't1_active_t': -1.26, 't1_cost_yr': 21.6, 't5_net_sharpe': 0.558, 't5_net_cagr': 6.4, 't5_active_bps': -2.1, 't5_active_t': -1.5, 'null_mean_t': -0.16, 'null_sd_t': 1.0, 'null_fire': 1, 'planted_t': 2.99, 'planted_welch': 4.67, 'planted_active_t': 3.08, 'planted_net_sharpe': 0.52, 'planted_bh_sharpe': -1.01}

## The signal — does the credit trend discriminate the day-t equity excess?

Split `xs = r_SPY − r_IEF` by the credit trend (HYG excess over IEF) known at the close of `t−1`. A positive trend should precede a *higher* `xs`.

In [2]:
print(f"126d (6mo): risk-on {R['on_bps']:+.2f} vs risk-off {R['off_bps']:+.2f} bps  "
      f"-> diff {R['diff_bps']:+.2f} bps/day  NW(10) t = {R['t_nw']:+.2f}  Welch t = {R['welch_t']:+.2f}")
print(f" 63d (3mo): diff {R['diff3_bps']:+.2f} bps/day  NW t = {R['t3_nw']:+.2f}")
print(f"risk-on share: {R['on_frac']*100:.1f}%% of days (the timer is mostly long SPY anyway)")

126d (6mo): risk-on +2.99 vs risk-off +4.60 bps  -> diff -1.62 bps/day  NW(10) t = -0.37  Welch t = -0.32
 63d (3mo): diff -2.55 bps/day  NW t = -0.61
risk-on share: 63.6%% of days (the timer is mostly long SPY anyway)


## Placebo — circular-shift the risk-on labels (1,000 draws)

In [3]:
print(f"observed {R['placebo_obs']:+.2f} bps vs placebo mean {R['placebo_mean']:+.2f} "
      f"(sd {R['placebo_sd']:.2f}) -> {R['placebo_sigma']:+.1f}sigma, p = {R['placebo_p']:.3f}")
print('the observed difference sits INSIDE the shuffle cloud -> nothing to explain')

observed -1.62 bps vs placebo mean +0.26 (sd 4.67) -> -0.4sigma, p = 0.646
the observed difference sits INSIDE the shuffle cloud -> nothing to explain


## Robustness — two eras (split 2017-01-01)

In [4]:
print(f"2007-2016 (n={R['era_early_n']}): diff {R['era_early_diff']:+.2f} bps  NW t = {R['era_early_t']:+.2f}")
print(f"2017-2026 (n={R['era_late_n']}): diff {R['era_late_diff']:+.2f} bps  NW t = {R['era_late_t']:+.2f}")
print('the sign FLIPS across eras -> an unstable non-signal')

2007-2016 (n=2310): diff +4.80 bps  NW t = +0.81
2017-2026 (n=2385): diff -12.27 bps  NW t = -1.81
the sign FLIPS across eras -> an unstable non-signal


## The timer — SPY when trend>0 else IEF, costed, vs 100%-SPY buy-and-hold

A switch turns the book over (2 legs); long-only, no borrow. 201 switches over 19 years.

In [5]:
for tag,ns,cg,ac,at in [('1 bp',R['t1_net_sharpe'],R['t1_net_cagr'],R['t1_active_bps'],R['t1_active_t']),
                        ('5 bps',R['t5_net_sharpe'],R['t5_net_cagr'],R['t5_active_bps'],R['t5_active_t'])]:
    print(f"{tag:>5}/leg: net Sharpe {ns:.3f} (B&H {R['t1_bh_sharpe']:.3f}) | net CAGR {cg:+.2f}%% "
          f"(B&H {R['t1_bh_cagr']:+.2f}%%) | active {ac:+.2f} bps/day (NW t={at:+.2f})")
print(f"drawdown: timer {R['t1_net_dd']:.0f}%% vs B&H {R['t1_bh_dd']:.0f}%% (the ONLY real benefit)")

 1 bp/leg: net Sharpe 0.627 (B&H 0.619) | net CAGR +7.32%% (B&H +10.87%%) | active -1.76 bps/day (NW t=-1.26)
5 bps/leg: net Sharpe 0.558 (B&H 0.619) | net CAGR +6.40%% (B&H +10.87%%) | active -2.10 bps/day (NW t=-1.50)
drawdown: timer -41%% vs B&H -55%% (the ONLY real benefit)


## Synthetic positive control — the machinery is unbiased

Live: the detector must NOT fire on the null and must recover a planted credit-times-equity relation. No network.

In [6]:
import os, sys
sys.path.insert(0, os.path.abspath('..'))
sys.path.insert(0, os.path.abspath(os.path.join('..','..','..')))
import numpy as np
from hy_credit_momentum import data, strategy as st
null_t = np.array([st.synthetic_detect(data.synthetic_panel(edge=0.0, seed=832+s, n_days=2200))['t_nw'] for s in range(8)])
print(f"null (edge=0), 8 seeds: NW t mean {null_t.mean():+.2f} (sd {null_t.std(ddof=1):.2f}), |t|>=2 in {(abs(null_t)>=2).sum()}/8")
planted = st.synthetic_detect(data.synthetic_panel(edge=0.006, seed=832, n_days=2200))
print(f"planted (edge=0.006): discrimination NW t = {planted['t_nw']:+.2f}, active NW t = {planted['active_t_nw']:+.2f}, "
      f"timer Sharpe {planted['net_sharpe']:+.2f} vs B&H {planted['bh_sharpe']:+.2f}")

null (edge=0), 8 seeds: NW t mean +0.33 (sd 1.16), |t|>=2 in 1/8
planted (edge=0.006): discrimination NW t = +2.99, active NW t = +3.08, timer Sharpe +0.52 vs B&H -1.01


## Verdict

- **Signal — None.** The credit trend does not time equity risk-on/off: the risk-on−risk-off equity-excess difference is wrong-signed and negligible (**-1.62 bps/day**, NW *t* = **-0.37** at 6 months; -2.55 bps, *t* = -0.61 at 3 months), sits -0.4σ inside a label-shift placebo (p = 0.65), and flips sign across eras (+4.8 → -12.3 bps). The 20-seed synthetic control recovers a *planted* effect (*t* = +2.99, fires on 1/20 nulls), so the flat real-tape result is a true absence.
- **Tradability — Mirage.** The costed SPY↔IEF timer never robustly beats buy-and-hold (net Sharpe 0.627 vs 0.619 at 1 bp, active NW *t* = -1.26; it forfeits 3.5%/yr of CAGR) and falls below B&H by 5 bps. Its one merit — a -41% vs -55% drawdown — is insurance paid for in forgone return, not an edge.